csharp/ql/test/utils/model-generator/theorems/TheoremSummaries.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Summaries;
+
+public class Theorems1<T> {
+    
+    public T Prop {
+        get { throw null; }
+        set { throw null; }
+    }
+
+    public Theorems1(T t)  { throw null; } 
+
+    public T Get() { throw null; }
+
+    public T Get(object x) { throw null; }
+
+    public T Id(T x) { throw null; }
+
+    public T Transform(int x, T y) { throw null; }
+
+    public S Transform<S>(S x) { throw null; }
+
+    public void Set(T x) { throw null; }
+
+    public void Set(int x, T y) { throw null; }
+
+    // No summary as S is unrelated to T
+    public void Set<S>(S x) { throw null; }
+
+    public int Apply(Func<T, int> f) { throw null; }
+}
+
+public class Theorems2 {
+
+    public T Transform<T>(T x) { throw null; }
```

Comment:
I'm not sure this is a case we'll want/need to cover.

Reply:
I will remove it, but I am not entirely sure that it is not needed.
If something has a type signature like this, then it could likely be the case that there are some type constraints on T that makes it reasonable to assume that flow is propagated from argument to return - either because a new value of type T is generated based on the input or that the existing value of T is manipulated by changing its state. One could imagine a generic with a type constraint `T : I`, where I is an interface with members that has side effects on the value of T when being called.

Reply:
But if `T` is constrained in that way, then I'm not sure we'll want the theorems-for-free approach at all - the theoretical foundation certainly becomes much weaker.

Reply:
Alright. Let's leave it out then :-)